pywhy_stats/discrepancy/bregman.py

```diff
@@ -0,0 +1,205 @@
+"""Bregman (conditional) discrepancy test.
+
+Also known as a conditional k-sample test, where the null hypothesis is that the
+conditional distributions are equal across different population groups. The
+Bregman tests for conditional divergence using correntropy.
+
+Returns
+-------
+PValueResult
+    The result of the test, which includes the test statistic and pvalue.
+"""
+
+from typing import Callable, Optional, Tuple
+
+import numpy as np
+from numpy.typing import ArrayLike
+from sklearn.base import BaseEstimator
+from sklearn.preprocessing import LabelBinarizer
+
+from pywhy_stats.kernel_utils import _preprocess_kernel_data, corrent_matrix, von_neumann_divergence
+
+from ..pvalue_result import PValueResult
+from .base import _compute_propensity_scores, _preprocess_propensity_data, compute_null
+
+
+def condind(
+    X: ArrayLike,
+    Y: ArrayLike,
+    group_ind: ArrayLike,
+    kernel: Optional[Callable[[ArrayLike], ArrayLike]] = None,
+    null_sample_size: int = 1000,
+    normalize_data: bool = False,
+    propensity_model=None,
+    propensity_weights=None,
+    centered: bool = False,
+    n_jobs: Optional[int] = None,
+    random_seed: Optional[int] = None,
+) -> PValueResult:
+    """
+    Test whether Y conditioned on X is invariant across the groups.
+
+    For testing conditional independence on continuous data, we compute Bregman divergences
+    :footcite:`Yu2020Bregman`. This specifically
+    tests the (conditional) invariance null hypothesis :math::
+
+        P_{Z=1}(Y|X) = P_{Z=0}(Y|X)
+
+    Parameters
+    ----------
+    X : ArrayLike of shape (n_samples, n_features_x)
+        Data for variable X, which can be multidimensional.
+    Y : ArrayLike of shape (n_samples, n_features_y)
+        Data for variable Y, which can be multidimensional.
+    group_ind : ArrayLike of shape (n_samples,)
```

Comment:
This basically assigns each sample in `X` and `Y` to either group 0 or 1, right? The docstring can be more explicit in this regard.

I am also wondering it wouldn't be clearer if we just accept four inputs: `X_baseline, X_reference, Y_baseline, Y_reference`, where baseline is `Z=0` and reference `Z=1` (or something along the line). This would definitely inflate the number of input arguments, but also don't need the ground indices anymore.

Reply:
I use `group_ind` because this would allow us to generalize these tests to `k-sample` in the future. I know some groups are working on this. I suppose we could for now do the 4 inputs, but later on if we generalize to k-sample, then we need to refactor to this design.

> This basically assigns each sample in `X` and `Y` to either group 0 or 1, right? The docstring can be more explicit in this regard.

Yep. Fixed the docstring.